src/profile/external-identities/OAuth2Account.tsx

```diff
@@ -51,93 +51,66 @@ const styles = {
   },
 };
 
-interface OAuth2LinkProps {
+interface OAuth2AccountProps {
   queryParams: { state?: string; code?: string };
   provider: OAuth2Provider;
 }
-export const OAuth2Link = (props: OAuth2LinkProps) => {
+export const OAuth2Account = (props: OAuth2AccountProps) => {
   const {
     queryParams: { state, code },
     provider,
   } = props;
 
+  const { oAuth2AccountStatus: storedOAuth2AccountStatus } = useStore(authStore);
+
+  const { externalUserId, expirationTimestamp } = storedOAuth2AccountStatus[provider.key] ?? {
+    externalUserId: undefined,
+    expirationTimestamp: undefined,
+  };
   const signal = useCancellation();
-  const [accountInfo, setAccountInfo] = useState<EcmLinkAccountResponse>();
-  const [accountLoaded, setAccountLoaded] = useState<boolean>(false);
```

Comment:
accounts are now loaded when the app detects the user has signed in, not when the user navigates to the External Identities page. Fence Accounts did this, but now all OAuth2 Accounts, including RAS and GitHub, will be loaded when the user signs in. This also means that RAS and GitHub links take advantage of the link expiration notifications that previously, only NIH Accounts and Fence Accounts utilized.